### PR TITLE
WIP: Support 1w, 1y for retention and table period

### PR DESCRIPTION
**Pull request by gouthamve**
This will let us `1w` in the schema config YAML and also lets us set the retention as `2y` etc. If this looks good, I'll go ahead and change docs, etc.

### CODE REVIEW

File: pkg/chunk/aws/dynamodb_table_client_test.go [hunk 1]
```diff
@@ -20,8 +20,8 @@ import (
 const (
 	tablePrefix      = "cortex_"
 	chunkTablePrefix = "chunks_"
-	tablePeriod      = 7 * 24 * time.Hour
-	gracePeriod      = 15 * time.Minute
+	tablePeriod      = model.Duration(7 * 24 * time.Hour)
+	gracePeriod      = model.Duration(15 * time.Minute)
```
Comment (bboreham) [inline pkg/chunk/aws/dynamodb_table_client_test.go]: As far as I can see this is always cast to a `time.Duration` so why not leave it as that?

File: pkg/chunk/table_manager.go [hunk 2]
```diff
@@ -115,9 +115,14 @@ type ProvisionConfig struct {
 func (cfg *TableManagerConfig) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.ThroughputUpdatesDisabled, "table-manager.throughput-updates-disabled", false, "If true, disable all changes to DB capacity")
 	f.BoolVar(&cfg.RetentionDeletesEnabled, "table-manager.retention-deletes-enabled", false, "If true, enables retention deletes of DB tables")
-	f.DurationVar(&cfg.RetentionPeriod, "table-manager.retention-period", 0, "Tables older than this retention period are deleted. Note: This setting is destructive to data!(default: 0, which disables deletion)")
+
+	cfg.RetentionPeriod = model.Duration(0)
```
Comment (bboreham) [inline pkg/chunk/table_manager.go]: why is this required?

**Comment by gouthamve** [comment]
Closing in favor of #2252
